### PR TITLE
Update documentation / demos to use Woodwork

### LINKED ISSUE #1291: Update pipeline/component documentation to use DataTables in all examples
Design doc for epic here: https://alteryx.quip.com/Oa4aA6h0Rf2k/Woodwork-DataTable-Integration-Design-Document

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1291 

Note: more work on updating graphing utils to support WW will be done in #1292; this PR only tracks updating what is necessary for our docs.

### CODE REVIEW

File: docs/source/demos/fraud.ipynb [hunk 1]
```diff
@@ -78,24 +78,26 @@
...
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "X = X.drop(['datetime', 'expiration_date', 'country', 'region', 'provider'], axis=1)\n",
+    "cols_to_drop = ['datetime', 'expiration_date', 'country', 'region', 'provider']\n",
+    "for col in cols_to_drop:\n",
+    "    X.pop(col)\n",
```
Comment (angela97lin) [inline docs/source/demos/fraud.ipynb:96]: Currently no `drop` method in Woodwork, so we pop each column from the DataTable :d

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 2]
```diff
@@ -194,23 +193,22 @@ def raise_zero_division(input_features):
...
-def test_explain_predictions_value_errors(input_features):
+def test_explain_predictions_value_errors():
     with pytest.raises(ValueError, match="Parameter input_features must be a non-empty dataframe."):
-        explain_predictions(None, input_features)
+        explain_predictions(None, pd.DataFrame())
 
 
 def test_output_format_checked():
-    input_features, y_true = pd.DataFrame({"a": range(15)}), pd.Series(range(15))
+    input_features, y_true = pd.DataFrame(data=[range(15)]), pd.Series(range(15))
```
Comment (angela97lin) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:202]: `pd.DataFrame({"a": range(15)})` produces a DataFrame with one column, but updated since the error text says we want a DataFrame with one row.

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 3]
```diff
@@ -169,9 +169,8 @@ def test_error_metrics():
                                    pd.Series([-np.log(0.9), -np.log(0.2)]))
 
 
-input_features_and_y_true = [([1], None, "^Input features must be a dataframe with more than 10 rows!"),
-                             (pd.DataFrame({"a": [1]}), None, "^Input features must be a dataframe with more than 10 rows!"),
-                             (pd.DataFrame({"a": range(15)}), [1], "^Parameter y_true must be a pd.Series."),
+input_features_and_y_true = [([[1]], pd.Series([1]), "^Input features must be a dataframe with more than 10 rows!"),
+                             (pd.DataFrame({"a": [1]}), pd.Series([1]), "^Input features must be a dataframe with more than 10 rows!"),
```
Comment (angela97lin) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:173]: Having invalid parameters will now raise an error in the Woodwork conversion code instead, so will never reach these. I think this is okay, given our docstrings explicitly ask for pd.Series/ww.DataColumn.

File: evalml/preprocessing/utils.py [hunk 4]
```diff
@@ -50,10 +55,11 @@ def split_data(X, y, regression=False, test_size=.2, random_state=None):
...
-        X = pd.DataFrame(X)
-    if not isinstance(y, pd.Series):
-        y = pd.Series(y)
+    X = _convert_to_woodwork_structure(X)
+    y = _convert_to_woodwork_structure(y)
+
+    X = _convert_woodwork_types_wrapper(X.to_dataframe())
+    y = _convert_woodwork_types_wrapper(y.to_series())
 
     if regression:
```
Comment (angela97lin) [inline evalml/preprocessing/utils.py:64]: `ShuffleSplit` runs into the same errors when passed Woodwork, so converting to pandas is necessary :(

File: docs/source/demos/fraud.ipynb [hunk 5]
```diff
@@ -26,7 +26,7 @@
    "source": [
     "## Configure \"Cost of Fraud\" \n",
     "\n",
-    "To optimize the pipelines toward the specific business needs of this model, you can set your own assumptions for the cost of fraud. These parameters are\n",
+    "To optimize the pipelines toward the specific business needs of this model, we can set our own assumptions for the cost of fraud. These parameters are\n",
```
Comment (angela97lin) [inline docs/source/demos/fraud.ipynb:29]: For consistency, updated second person --> first person

File: evalml/demos/breast_cancer.py [hunk 6]
```diff
@@ -1,8 +1,9 @@
 import pandas as pd
+import woodwork as ww
 from sklearn.datasets import load_breast_cancer as load_breast_cancer_sk
 
 
-def load_breast_cancer():
+def load_breast_cancer(return_pandas=False):
```
Comment (angela97lin) [inline evalml/demos/breast_cancer.py:6]: I decided to go with adding a parameter because it'd still be nice to give users that flexibility, but lmk any thoughts!

File: docs/source/demos/text_input.ipynb [hunk 7]
```diff
@@ -89,7 +89,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "EvalML uses [Woodwork](https://woodwork.alteryx.com/en/stable/) to automatically detect which columns are text columns, so you can run search normally, as you would if there was no text data."
+    "EvalML uses [Woodwork](https://woodwork.alteryx.com/en/stable/) to automatically detect which columns are text columns, so you can run search normally, as you would if there was no text data. We can print out the logical type of the `Message` column and assert that it is indeed inferred as a natural language column."
```
Comment (dsherry) [inline docs/source/demos/text_input.ipynb:92]: Nice

File: docs/source/demos/text_input.ipynb [hunk 8]
```diff
@@ -121,7 +118,7 @@
     "                      max_batches=1,\n",
     "                      optimize_thresholds=True)\n",
     "\n",
-    "automl.search(X_train_dt, y_train_dc)"
+    "automl.search(X_train, y_train)"
```
Comment (dsherry) [inline docs/source/demos/text_input.ipynb:121]: Looks a lot better!

File: docs/source/demos/text_input.ipynb [hunk 9]
```diff
@@ -240,7 +239,7 @@
    "source": [
     "## Why encode text this way?\n",
     "\n",
-    "To demonstrate the importance of text-specific modeling, let's train a model with the same dataset, without letting `AutoMLSearch` detect the text column. We can change this by explicitly setting the data type of the `'Message'` column in Woodwork."
+    "To demonstrate the importance of text-specific modeling, let's train a model with the same dataset, without letting `AutoMLSearch` detect the text column. We can change this by explicitly setting the data type of the `'Message'` column in Woodwork to `Categorical`."
```
Comment (dsherry) [inline docs/source/demos/text_input.ipynb:242]: Nit pick but should "categorical" be lowercase? Maybe case doesn't matter for woodwork type matching, idk

**Review by dsherry — APPROVED** [review]
@angela97lin this is huge! Left some suggestions but pretty much ready to 🚢

File: evalml/preprocessing/utils.py (hunk 4 above)
Comment (angela97lin) [reply evalml/preprocessing/utils.py:63]: @dsherry Ah right, we can use `iloc` now, so `train, test = next(CV_method.split(X.to_dataframe(), y.to_series()))` works. I think `index` works somewhat differently in Woodwork, since it was set to None for our test data 🤔

File: docs/source/demos/text_input.ipynb (hunk 9 above)
Comment (angela97lin) [reply docs/source/demos/text_input.ipynb:242]: Ah I've always just used "Categorical" since that's what the [actual name](https://woodwork.alteryx.com/en/stable/_modules/woodwork/logical_types.html#Categorical) is but you're right, woodwork type matching doesn't matter 🤷

File: evalml/tests/objective_tests/test_binary_classification_objective.py [hunk 10]
```diff
@@ -45,6 +45,5 @@ def test_decision_function_neg():
     ypred_proba = np.arange(6) / 5.0
     y_true = pd.Series(np.array([0] * 3 + [1] * 3), dtype=bool)
     obj = F1()
-    pd.testing.assert_series_equal(obj.decision_function(ypred_proba.tolist()), y_true)
```
[the remark this answers is not in the document]
Comment (angela97lin) [reply evalml/tests/objective_tests/test_binary_classification_objective.py:48]: Current impl of woodwork wrapper doesn't support lists, tracking w/ #1456

File: evalml/demos/breast_cancer.py (hunk 6 above)
Comment (angela97lin) [reply evalml/demos/breast_cancer.py:6]: @dsherry Thanks! I looked at the link and am going to update this to 
```
Union[(ww.DataTable, ww.DataColumn), (pd.Dataframe, pd.Series)]: X and y
```

I think this makes a bit more sense because it's more clear that it either returns two Woodwork or two pandas, but lmk if you think otherwise!

Comment (dsherry) [reply evalml/demos/breast_cancer.py:6]: @angela97lin yep that looks good! Our sphinx API ref gen doesn't do any sort of parsing of return type currently, so I'm on board with whatever is communicative and more or less in-line with the spec :)

**Comment by angela97lin** [comment]
(Verified docs locally since RtD is still broken)
